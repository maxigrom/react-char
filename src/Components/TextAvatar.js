// @flow

import React from 'react';
import Avatar from '@material-ui/core/Avatar';
import { getColor } from '../utils/getColor';

type Props = {
  className?: string,
  value: string,
}

const TextAvatar = (props: Props) => (
  <Avatar alt={props.value} className={props.className} style={{ backgroundColor: getColor(props.value)}}>
    {props.value.substring(0, 2)}
  </Avatar>
);

export default TextAvatar
