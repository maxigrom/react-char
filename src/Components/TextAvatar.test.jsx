/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TextAvatar from './TextAvatar';

describe('<TextAvatar />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TextAvatar value="Max Attacks" />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<TextAvatar value="Max Attacks" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
