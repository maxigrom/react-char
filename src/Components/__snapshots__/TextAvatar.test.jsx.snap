// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextAvatar /> renders correctly 1`] = `
<div
  className="MuiAvatar-root-4 MuiAvatar-colorDefault-5"
  style={
    Object {
      "backgroundColor": "#e91e63",
    }
  }
>
  Ma
</div>
`;
